### ISSUE TITLE
Proxy cache expanded support for additional registries

### ISSUE THREAD

**Issue by xaleeks** [goharbor/harbor#12885]
Support harbor acting as pull through cache for more registries beyond harbor and Dockerhub.

**Comment by liangminhua** [comment]
please support docker registry

**Comment by ChristianCiach** [comment]
We have a local docker-registry that is based on `library/registry:2`. If I lie to harbor and configure this registry using "Provider: Harbor", the resulting proxy-cache works just fine. 

So, it seems that "docker registry" as proxy-cache is already working. Is there some technical limitation that I don't know of?

**Comment by ChristianCiach** [comment]
We have a local docker registry that is based on the image `library/registry:2` from docker hub. If I lie to Harbor and configure this registry using "Provider: Harbor", the resulting proxy-cache seems to work just fine.

**Comment by xaleeks** [comment]
@ChristianCiach  I'm not sure I understand, what do you mean by 'based on the image from docker hub'.  You are configuring a dockerhub endpoint as a harbor type endpoint?

**Comment by ChristianCiach** [comment]
@xaleeks No, a "dockerhub endpoint" can only download images from docker hub (meaning docker.io). What I am trying to say is that we are using a private registry that is based on this image: https://hub.docker.com/_/registry

To use harbor as a proxy cache for our private registry that is based on this image, I can configure a "harbor endpoint" in harbor, pointing it to our private registry, and it seems to work just fine.

**Comment by xaleeks** [comment]
That sounds correct then, since Harbor is just running docker distribution on the inside.

**Comment by xaleeks** [comment]
I'm expecting GCR, Quay.io, ECR, ACR in that order of importance.  Let me know if you disagree @steven-zou  @stonezdj

**Comment by starkoff** [comment]
Unfortunately. Not append  in  "New Registry Endpoint"  Docker Hub.  Message "Failed to ping endpoint."  I can't get ping from hub.docker.com , but docker or docker compose works correctly.

**Comment by gw0** [comment]
Don't forget GitLab Container Registry, GitHub Container Registry, and already mentioned self-hosted Docker Registry 2.0.

**Comment by olfway** [comment]
up

**Comment by hinrichd** [comment]
It would be nice to have also support for  GitLab Container Registry as proxy cache, too! THX

**Comment by XLordalX** [comment]
+1 GitLab container registry

**Comment by Kovah** [comment]
Now that Docker Hub has launched their [Hardened Images](https://www.docker.com/products/hardened-images/), this feature is even more important. I'm pretty sure that their Hardened Images registry works in the same way as their Hub registry, but just uses a different domain (dhi.io instead of docker.io)

As long as Harbor does not offer support for that new registry, users would need to authenticate and pull directly, which comes with quite a few downsides.

**Comment by ChristianCiach** [comment]
@Kovah dhi.io works just fine when configuring the registry with type `Docker Registry`.
